zipkin-storage/dse5/src/main/java/zipkin/storage/cassandra3/Dse5Schema.java

```diff
@@ -0,0 +1,332 @@
+    }
+    return result;
+  }
+
+  static void applyCqlFile(String keyspace, Session session, String resource) {
+    try (Reader reader = new InputStreamReader(Dse5Schema.class.getResourceAsStream(resource))) {
+      for (String cmd : CharStreams.toString(reader).split(";")) {
+        cmd = cmd.trim().replace(" " + DEFAULT_KEYSPACE, " " + keyspace);
+        if (!cmd.isEmpty()) {
+          session.execute(cmd);
+        }
+      }
+    } catch (IOException ex) {
+      LOG.error(ex.getMessage(), ex);
+    }
+  }
+
+  @UDT(keyspace = DEFAULT_KEYSPACE + "_udts", name = "endpoint")
+  static final class EndpointUDT {
+
+    private String service_name;
+    private InetAddress ipv4;
+    private InetAddress ipv6;
+    private Short port;
+
+    EndpointUDT() {
+      this.service_name = null;
+      this.ipv4 = null;
+      this.ipv6 = null;
+      this.port = null;
+    }
+
+    EndpointUDT(Endpoint endpoint) {
+      this.service_name = endpoint.serviceName;
+      this.ipv4 = endpoint.ipv4 == 0 ? null : InetAddresses.fromInteger(endpoint.ipv4);
+      if (endpoint.ipv6 != null && endpoint.ipv6.length == 16) {
+        try {
+          this.ipv6 = Inet6Address.getByAddress(endpoint.ipv6);
+        } catch (UnknownHostException ex) { // We already checked for illegal length, so unexpected!
+        }
+      }
+      this.port = endpoint.port;
+    }
+
+    public String getService_name() {
+      return service_name;
+    }
+
+    public InetAddress getIpv4() {
+      return ipv4;
+    }
+
+    public InetAddress getIpv6() {
+      return ipv6;
+    }
+
+    public Short getPort() {
+      return port;
+    }
+
+    public void setService_name(String service_name) {
+      this.service_name = service_name;
+    }
+
+    public void setIpv4(InetAddress ipv4) {
+      this.ipv4 = ipv4;
+    }
+
+    public void setIpv6(InetAddress ipv6) {
+      this.ipv6 = ipv6;
+    }
+
+    public void setPort(short port) {
+      this.port = port;
+    }
+
+    private Endpoint toEndpoint() {
+      Endpoint.Builder builder = Endpoint.builder()
+          .serviceName(service_name)
+          .ipv4(ipv4 == null ? 0 : ByteBuffer.wrap(ipv4.getAddress()).getInt())
+          .port(port);
+
+      if (null != ipv6) {
+        builder = builder.ipv6(ipv6.getAddress());
+      }
+      return builder.build();
+    }
+  }
+
+  @UDT(keyspace = DEFAULT_KEYSPACE + "_udts", name = "annotation")
+  static final class AnnotationUDT {
+
+    private long ts;
+    private String v;
+    private EndpointUDT ep;
+
+    AnnotationUDT() {
+      this.ts = 0;
+      this.v = null;
+      this.ep = null;
+    }
+
+    AnnotationUDT(Annotation annotation) {
+      this.ts = annotation.timestamp;
+      this.v = annotation.value;
+      this.ep = annotation.endpoint != null ? new EndpointUDT(annotation.endpoint) : null;
+    }
+
+    public long getTs() {
+      return ts;
+    }
+
+    public String getV() {
```

Comment:
Minor readability suggestion (@adriancole probably can comment more) but I prefer to not have abbreviations that are not generally well known in public APIs. e.g `ep` probably don't mean much to most people